File: problem/bry/C++/5. Longest Palindromic Substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int n = s.length();
        pair <int, int> ans= make_pair(0,0);
        for(int i = 0 ; i < n ; ++i){
            int l, r;
            l = r = i;
            while(l-1 >= 0 && s[l-1] == s[i]) --l;
            while(r+1 < n && s[r+1] == s[i]) ++r;
            while(l-1 >= 0 && r+1 < n && s[l-1] == s[r+1]){
                --l;
                ++r;
            }
            ans =  (r-l+1) > ans.second ? make_pair(l, mx) : ans;
        }
        return s.substr(lm, mx);
    }
};

File: problem/bry/Kotlin/5. Longest Palindromic Substring.kt
class Solution {
    fun longestPalindrome(s: String): String {
        var ans = Triple(0, 0, 0)
        val n = s.length
        for (i in 0 until n){
            var (r, l) = Pair(i, i)
            while(r+1 < n && s[r+1] == s[i]) ++r;
            while(l-1 >= 0 && s[l-1] == s[i]) --l;
            while(r+1 < n && l-1 >= 0 && s[r+1] == s[l-1]){
                ++r;
                --l;
            }
            ans = if (r-l+1 > ans.third) Triple(l, r, r-l+1) else ans
        }
        return s.subSequence(ans.first, ans.second+1).toString()
    }
}